tests/interpreters/test_eqsat_pdl_interpreter.py

```diff
@@ -0,0 +1,393 @@
+from xdsl.dialects import pdl, test
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    IntegerAttr,
+    IntegerType,
+    StringAttr,
+    i32,
+    i64,
+)
+from xdsl.interpreters.eqsat_pdl import EqsatPDLMatcher
+from xdsl.ir import Block
+from xdsl.utils.test_value import TestSSAValue
+
+
+def test_match_type():
+    matcher = EqsatPDLMatcher()
+
+    pdl_op = pdl.TypeOp()
+    ssa_value = pdl_op.result
+    xdsl_value = StringAttr("a")
+
+    # New value
+    assert matcher.match_type(ssa_value, pdl_op, xdsl_value)
+    assert matcher.matching_context == {ssa_value: xdsl_value}
+
+    # Same value
+    assert matcher.match_type(ssa_value, pdl_op, xdsl_value)
+    assert matcher.matching_context == {ssa_value: xdsl_value}
+
+    # Other value
+    assert not matcher.match_type(ssa_value, pdl_op, StringAttr("b"))
+    assert matcher.matching_context == {ssa_value: xdsl_value}
```

Comment:
Since only match_operand is overridden, it seems like all the other unit tests are not needed, since the other functions are already tested? Could you please remove all the tests other than `test_match_operand`, and add the case to check that `NotImplementedError` is raised as expected?

Reply:
Sorry, just saw your comments - I made some changes and it now passed the tests... what do you think?

Reply:
Looks great!